TOMOYO Linux uses pathnames for auditing and controlling file access.
Therefore, namespace_sem is needed.

Signed-off-by: Kentaro Takeda <takedakn@nttdata.co.jp>
Signed-off-by: Tetsuo Handa <penguin-kernel@I-love.SAKURA.ne.jp>
---
 fs/namespace.c                |    2 +-
 include/linux/mnt_namespace.h |    2 ++
 2 files changed, 3 insertions(+), 1 deletion(-)

--- linux-2.6.orig/fs/namespace.c	2007-08-23 21:25:13.000000000 +0900
+++ linux-2.6/fs/namespace.c	2007-08-24 15:51:34.000000000 +0900
@@ -38,7 +38,7 @@ static int event;
 static struct list_head *mount_hashtable __read_mostly;
 static int hash_mask __read_mostly, hash_bits __read_mostly;
 static struct kmem_cache *mnt_cache __read_mostly;
-static struct rw_semaphore namespace_sem;
+struct rw_semaphore namespace_sem;
 
 /* /sys/fs */
 decl_subsys(fs, NULL, NULL);
--- linux-2.6.orig/include/linux/mnt_namespace.h	2007-08-23 21:25:13.000000000 +0900
+++ linux-2.6/include/linux/mnt_namespace.h	2007-08-24 15:51:34.000000000 +0900
@@ -6,6 +6,8 @@
 #include <linux/sched.h>
 #include <linux/nsproxy.h>
 
+extern struct rw_semaphore namespace_sem;
+
 struct mnt_namespace {
 	atomic_t		count;
 	struct vfsmount *	root;
